features/networking/multus.feature

```diff
@@ -416,4 +416,248 @@ Feature: Multus-CNI related scenarios
+      | resource | node                    |
+      | name     | <%= cb.nodes[0].name %> |
+      | key_val  | test-                   |
+    Then the step should succeed
+    """ 
+    #Creating first pod in vlan 100
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/generic_multus_pod_nodeselector.yaml" replacing paths:
+      | ["metadata"]["name"] | pod1-vlan100 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"]| bridgevlan100 |
+      | ["spec"]["nodeSelector"]["test"] | worker1 |
+    Then the step should succeed
+    And the pod named "pod1-vlan100" becomes ready
+    And evaluation of `pod.name` is stored in the :pod1 clipboard
+    And I execute on the pod:
+      | ifconfig | net1 |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/)[0]` is stored in the :pod1_net1_ip clipboard
+    
+    #Creating 2nd pod on same node as first in vlan 100
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/generic_multus_pod_nodeselector.yaml" replacing paths:
+      | ["metadata"]["name"] | pod2-vlan100 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"]| bridgevlan100 |
+      | ["spec"]["nodeSelector"]["test"] | worker1 |
+    Then the step should succeed
+    And the pod named "pod2-vlan100" becomes ready
+    And evaluation of `pod.name` is stored in the :pod2 clipboard
+    And I execute on the pod:
+      | ifconfig | net1 |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/)[0]` is stored in the :pod2_net1_ip clipboard
+    
+    #Creating 3rd pod on same node but in vlan 200
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/generic_multus_pod_nodeselector.yaml" replacing paths:
+      | ["metadata"]["name"] | pod3-vlan200 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"]| bridgevlan200 |
+      | ["spec"]["nodeSelector"]["test"] | worker1 |
+    Then the step should succeed
+    And the pod named "pod3-vlan200" becomes ready
+    And evaluation of `pod.name` is stored in the :pod3 clipboard
+    And I execute on the pod:
+      | ifconfig | net1 |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/)[0]` is stored in the :pod3_net1_ip clipboard
+    
+    #Clean-up required to erase bridge interfcaes created on node
+    Given I register clean-up steps:
+    """
+    the bridge interface named "mybridge" is deleted from the "<%= cb.nodes[0].name %>" node
+    the bridge interface named "mybridge.100" is deleted from the "<%= cb.nodes[0].name %>" node
+    the bridge interface named "mybridge.200" is deleted from the "<%= cb.nodes[0].name %>" node
+    """  
```

Comment:
It appears to me that these bridges are created earlier, so could you move clean-up of bridges to the place immediately after they were created?

.. unless I am mistaken

Reply:
btw is it not a bug that we deleted project but bridges remain?

Reply:
i had this discussion with dev last month. As pr CNF design, we are not supposed to cleanup bridge interfaces on node due to possibility of then for reuse. But in test we need to clear them up manually else they will interfere with output of next similar test

Reply:
hmm, doesn't sound convincing to me but I'll pass this time

Reply:
Still registering clean-up would make sense for your scenarios immediately before/after place where these are created.

Reply:
Ya i was not convinced too :( https://bugzilla.redhat.com/show_bug.cgi?id=1768289
About cleanup steps, i will take a look. If cleanup is placed after pod creation then its fine because bridge actually gets created on a node when a pod gets created and landed up on that node

Reply:
There is no harm if you put the delete step even before creating the resource. The step logic is to ignore if object doesn't exist. Only fail if object exists and delete fails.
If there is any issue with pod and it doesn't start properly but bridge is created, registering later may never happen.
That's why I suggest moving earlier.

Reply:
fixed. Adjusted in more sensible way